trl/trainer/grpo_trainer.py

```diff
@@ -1086,12 +1086,30 @@ def _compute_loss(self, model, inputs):
             mean_kl = (per_token_kl * completion_mask).sum() / completion_mask.sum()
             self._metrics[mode]["kl"].append(self.accelerator.gather_for_metrics(mean_kl).nanmean().item())
 
-        # Compute the clip ratio
-        is_clipped = ((coef_1 < 1 - self.epsilon_low) & (advantages.unsqueeze(1) < 0)) | (
-            (coef_1 > 1 + self.epsilon_high) & (advantages.unsqueeze(1) > 0)
+        # Compute the clipped probability ratios
+        is_low_clipped = (coef_1 < 1 - self.epsilon_low) & (advantages.unsqueeze(1) < 0)
+        is_high_clipped = (coef_1 > 1 + self.epsilon_high) & (advantages.unsqueeze(1) > 0)
+        is_region_clipped = is_low_clipped | is_high_clipped
+
+        low_clip = (is_low_clipped * completion_mask).sum() / completion_mask.sum()
+        high_clip = (is_high_clipped * completion_mask).sum() / completion_mask.sum()
+        clip_ratio = (is_region_clipped * completion_mask).sum() / completion_mask.sum()
+
+        self._metrics[mode]["clip_ratio/low_mean"].append(
+            self.accelerator.gather_for_metrics(low_clip).nanmean().item()
+        )
+        self._metrics[mode]["clip_ratio/low_min"].append(
+            torch.min(self.accelerator.gather_for_metrics(low_clip)).item()
+        )
+        self._metrics[mode]["clip_ratio/high_mean"].append(
+            self.accelerator.gather_for_metrics(high_clip).nanmean().item()
+        )
+        self._metrics[mode]["clip_ratio/high_max"].append(
+            torch.max(self.accelerator.gather_for_metrics(high_clip)).item()
```

Comment:
Some values in the gathered `high_clip` can be `nan`, (eg, because of the overlong filtering). We need a `nanmax` here that would ignore these values. Currently it would work, but as soon as you have one nan, the logged value is nan.
I'll push a fix so that you can merge tomorrow

Reply:
Red: before e6d013f
Green: after e6d013f

<img width="1050" alt="Screenshot 2025-04-15 at 14 39 51" src="https://github.com/user-attachments/assets/6565876b-4608-4c9e-a14d-646fda5c9a1c" />

Reply:
```python
from datasets import load_dataset
from trl import GRPOTrainer, GRPOConfig
import random

dataset = load_dataset("trl-lib/tldr", split="train[:200]")


def reward_random(completions, **kwargs):
    return [random.random() for _ in range(len(completions))]

trainer = GRPOTrainer(
    model="Qwen/Qwen2.5-1.5B",
    reward_funcs=reward_random,
    train_dataset=dataset,
    args=GRPOConfig(
        num_iterations=4,
        per_device_train_batch_size=2,
        mask_truncated_completions=True,
        logging_steps=10,
    ),
)
trainer.train()
```

4 GPUs

```sh
accelerate launch 3289.py
```

Reply:
Very nice fix, thanks!